cristin-import/src/main/java/no/unit/nva/cristin/lambda/CristinPatchEventConsumer.java

```diff
@@ -114,17 +126,19 @@ private UriWrapper constructSuccessFileUri(Publication publication, EventReferen
             .addChild(publication.getIdentifier() + JSON);
     }
 
-    private Optional<Publication> processMessage(SQSMessage message) {
+    private void processMessage(SQSMessage message) {
         logger.info("received message: " + message.getBody());
-        return attempt(() -> getEventReferenceFromBody(message.getBody())).map(this::processEvent)
-                   .toOptional();
+        attempt(() -> getEventReferenceFromBody(message.getBody())).map(this::processChild)
+            .toOptional();
+        attempt(() -> getEventReferenceFromBody(message.getBody())).map(this::processParent)
+            .toOptional();
```

Comment:
It has to be to different attempts, otherwise error in processChild() will jump over processParent().